R/reportDIETER.R

```diff
@@ -0,0 +1,284 @@
+    
+    outAnnual$Unit<- substring(outAnnual$Variable, regexpr("\\(", outAnnual$Variable))
+    outAnnual <-  outAnnual %>% dplyr::mutate_(Unit =~if_else(grepl("\\(",Unit),Unit,"NA") )
+    
+    outAnnual$Variable <- mapply(gsub, "\\(.*", "",outAnnual$Variable)
+    outAnnual$Unit <- mapply(gsub, "\\(", "",  outAnnual$Unit)
+    outAnnual$Unit <- mapply(gsub, "\\)", "",  outAnnual$Unit)
+    
+    
+    return(outAnnual)
+    
+  }
+  
+  output_folder <- outputDir
+  outAnnual <- gdxToQuitteAnnual(dieterDatafile)
+  annual_new <- data.table::dcast(outAnnual, ... ~ Year, value.var="Value")
+  
+  
+  # check if these years are already present, if not then add
+  if(!'2005' %in% colnames(annual_new)){
+    annual_new$`2005` <- NA
+  }
+  
+  if(!'2110' %in% colnames(annual_new)){
+    annual_new$`2110` <- NA
+  }
+  if(!'2130' %in% colnames(annual_new)){
+    annual_new$`2130`<- NA
+  }
+  if(!'2150' %in% colnames(annual_new)){
+    annual_new$`2150` <- NA
+  }
+  
+  
+  #technology mapping
+  dieter.tech.mapping <- c(CCGT = "CCGT",
+                           #lig = "Lignite",
+                           Solar = "Solar",
+                           Wind_on = "Wind",
+                           bio = "Biomass",
+                           OCGT_eff = "OCGT",
+                           ror = "Hydro",
+                           nuc = "Nuclear"
+                           #hc = "Hard coal",
+                           #coal = "Coal (Lig + HC)",
+  )
+  
+  
+  
+  dieter.tech.mapping.standalone <- c(dieter.tech.mapping,
+                                      lig = "Lignite",
+                                      hc = "Hard coal",
+                                      coal = "Coal (Lig + HC)",
+                                      NULL)
+  
+  annual_new$Tech <- as.factor(annual_new$Tech)
+  annual_remind <-  annual_new
+  annual_new <- annual_new %>% quitte::revalue.levels(Tech = dieter.tech.mapping.standalone)
+  
+  
+  annual_new$Variable <- paste0(stringr::str_trim(annual_new$Variable,side = c("both")),"|",annual_new$Tech)
+  annual_new  <- annual_new %>% dplyr::select_(~-Period,~-Tech,~-`2005`)
+  
+  
+  data.table::setDT(annual_new)
+  EOL <- if (.Platform$OS.type=="windows") ";\r\n" else ";\n"
+ 
+  dir.create(file.path(outputDir, subFolder), showWarnings = FALSE)
+  # save Dieter Data as a seperate mif & rds
+  fwrite(annual_new, file.path(outputDir, subFolder,"Dieter_Annual.mif"), append=F, sep=";", eol=EOL)
+  read.quitte(file.path(outputDir, subFolder,"Dieter_Annual.mif")) %>% 
+    readr::write_rds(paste0(gsub(".mif","",file.path(outputDir, subFolder,"Dieter_Annual.mif")),'.rds'), compress = 'xz') 
+  
+  
+  # append model = 'DIETER' to REMIND-EU mif file
+  # then for the switch that appends to the main mif, use coal = "Coal (Lig + HC)", 
+  
+  dieter.tech.mapping.REMIND <- c(dieter.tech.mapping,
+                                  coal = "Coal (Lig + HC)",
+                                  NULL)
```

Comment:
NULL is ignored by `c()`
```suggestion
                                  )
```